release notes/v0.31.0.md

```diff
@@ -0,0 +1,75 @@
+
+For example:
+
+```javascript
+import http from 'k6/http';
+
+// Set expected statuses globally for all requests.
+http.setResponseCallback(http.expectedStatuses({min: 200, max: 399}, 418));
+
+export default function () {
+  // This request will be marked as failed.
+  http.get('https://httpbin.test.k6.io/status/400');
+  // This request will be considered as "passed" because of the responseCallback override.
+  http.get('https://httpbin.test.k6.io/status/400', { responseCallback: http.expectedStatuses(400) });
+}
+```
+
+Running this script will produce a summary like:
+
+```
+http_req_duration..............: avg=204.57ms min=203.31ms med=204.57ms max=205.82ms p(90)=205.57ms p(95)=205.7ms
+  { expected_response:true }...: avg=203.31ms min=203.31ms med=203.31ms max=203.31ms p(90)=203.31ms p(95)=203.31ms
+http_req_failed................: 50.00% ✓ 1   ✗ 1
+```
+
+Note the new `http_req_duration` sub-metric for passed requests only and the `http_req_failed` Rate metric.
+
+The `http.setResponseCallback()` is meant to allow arbitrary JS functions to process responses, but for now only the `http.expectedStatuses()` callback is supported.
+
+
+## Other enhancements and UX improvements
+
+- JS: Because of the awesome improvements to [goja](https://github.com/dop251/goja), the JS runtime k6 uses, it's no longer necessary for k6 to load [core.js](https://github.com/zloirock/core-js) to polyfill missing JS features when using the default [`--compatibility-mode=extended`](https://k6.io/docs/using-k6/javascript-compatibility-mode). So in v0.31.0 core.js has been dropped entirely, yielding some significant CPU and memory usage improvements. The actual numbers will depend on the use case, but users can expect [TODO: Mention a percentage range?] ([#1824](https://github.com/loadimpact/k6/pull/1824))
+- JS: Also because of goja improvements, some Babel plugins were disabled which should decrease CPU and memory usage even more. ([#1822](https://github.com/loadimpact/k6/pull/1822))
+- JS: Expanded `ArrayBuffer` support in most internal modules, so now you can pass `ArrayBuffer` to `http.file()`, in `k6/encoding` and `k6/crypto` functions. This makes working with binary files more efficient as it doesn't require string translations. In upcoming versions we plan to expand this to the WebSocket module, as well as make some potentially breaking changes for APIs that currently return string (see the details in the Breaking Changes announcement below). ([#1800](https://github.com/loadimpact/k6/pull/1800))
+- The Docker image base was updated to Alpine 3.13. Thanks @andriisoldatenko! ([#1821](https://github.com/loadimpact/k6/pull/1821))
+- The Debian package now includes `ca-certificates` as a dependency. Thanks @Bablzz! ([#1854](https://github.com/loadimpact/k6/pull/1854))
+
+
+## Bugs fixed!
+
+- JS: Babel is now also run on compilation errors, which improves support of some obscure language features. ([#1861](https://github.com/loadimpact/k6/pull/1861))
+
```

Comment:
We probably should mention the concrete bugs with outputs that were fixed here - https://github.com/loadimpact/k6/issues/1423, https://github.com/loadimpact/k6/issues/1606, https://github.com/loadimpact/k6/issues/923

https://github.com/loadimpact/k6/issues/1820 should probably also be mentioned

Reply:
Should be resolved by afbcf9f1 and be8d9fdf.